dependencies/dex/dex.yaml

```diff
@@ -0,0 +1,124 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: dex
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: dex-cert
+  namespace: dex
+spec:
+  isCA: true
+  subject:
+    organizations:
+      - konflux
+  dnsNames:
+  - localhost
+  - dex.dex
+  issuerRef:
+    kind: ClusterIssuer
+    name: self-signed-cluster-issuer
+  secretName: dex-cert
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: dex
+  name: dex
+  namespace: dex
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: dex
+  template:
+    metadata:
+      labels:
+        app: dex
+    spec:
+      serviceAccountName: dex # This is created below
+      containers:
+      - image: ghcr.io/dexidp/dex:v2.32.0
+        name: dex
+        command: ["/usr/local/bin/dex", "serve", "/etc/dex/cfg/config.yaml"]
+        ports:
+        - name: https
+          containerPort: 9443
+        volumeMounts:
+        - name: dex
+          mountPath: /etc/dex/cfg
+        - name: tls
+          mountPath: /etc/dex/tls
+        readinessProbe:
+          httpGet:
+            path: /idp/healthz
```

Comment:
@gbenhaim should it be ok to have `idp` in the path here? It's defined in the nginx configs.

Reply:
you don't need to prefix the path with `/idp` since t he rediness probe doesn't go through the proxy, it contacts the container in the pod directly. `/healthz` should be enough.

Reply:
This is what I get without it:
```
Readiness probe failed: HTTP probe failed with statuscode: 404
```

Reply:
after looking at the dex helm chart and example config I think that you need to use another endpoint for health checks:

https://github.com/dexidp/dex/blob/master/config.yaml.dist#L68

https://github.com/dexidp/helm-charts/blob/master/charts/dex/templates/deployment.yaml#L110

Reply:
Thanks for that. I changed things accordingly.